src/main/java/tr/com/srdc/cda2fhir/transform/CCDTransformerImpl.java

```diff
@@ -92,122 +91,193 @@ public CCDTransformerImpl() {
+	 * 
+	 * @param filePath A file path string to a Consolidated CDA (C-CDA) 2.1
+	 *                 Continuity of Care Document (CCD) on file system
+	 * @return A FHIR Bundle that contains a Composition corresponding to the CCD
+	 *         document and all other resources that are referenced within the
+	 *         Composition.
+	 * @throws Exception
+	 */
+	public Bundle transformDocument(String filePath) throws Exception {
+		ContinuityOfCareDocument cda = getClinicalDocument(filePath);
+		return transformDocument(cda, true);
+	}
+
+	/**
+	 * Transforms a Consolidated CDA (C-CDA) 2.1 Continuity of Care Document (CCD)
+	 * instance to a Bundle of corresponding FHIR resources
+	 * 
+	 * @param cda                A Consolidated CDA (C-CDA) 2.1 Continuity of Care
+	 *                           Document (CCD) instance to be transformed
+	 * @param bundleType         The type of bundle to create, currently only
+	 *                           supports transaction bundles.
+	 * @param resourceProfileMap The mappings of default resource profiles to
+	 *                           desired resource profiles. Used to set profile
+	 *                           URI's of bundle entries or omit unwanted entries.
+	 * @param provenance         An optional FHIR provenance object that may be
+	 *                           inserted into the bundle.
+	 * @return A FHIR Bundle that contains a Composition corresponding to the CCD
+	 *         document and all other resources that are referenced within the
+	 *         Composition.
+	 * @throws Exception
+	 */
+	public Bundle transformDocument(ContinuityOfCareDocument cda, BundleType bundleType,
+			Map<String, String> resourceProfileMap, Provenance provenance) throws Exception {
+		Bundle bundle = transformDocument(cda, true);
+		bundle.setType(bundleType);
+		if (!provenance.equals(null)) {
+			// TODO: Code here to take object and add it bundle, and build provenance.target
+			// on it for all of the resources.
+			// https://www.hl7.org/fhir/stu3/provenance.html, look at section 6.3.4.6.
+			bundle.addEntry(new BundleEntryComponent().setResource(provenance));
+		}
+		if (bundleType.equals(BundleType.TRANSACTION)) {
+			return createTransactionBundle(bundle, resourceProfileMap, false);
+		}
+		return bundle;
+	}
+
+	/**
+	 * Transforms a Consolidated CDA (C-CDA) 2.1 Continuity of Care Document (CCD)
+	 * instance to a Bundle of corresponding FHIR resources
+	 * 
+	 * @param cda        A Consolidated CDA (C-CDA) 2.1 Continuity of Care Document
+	 *                   (CCD) instance to be transformed
+	 * @param provenance An optional FHIR provenance object that may be inserted
+	 *                   into the bundle.
+	 * @return A FHIR Bundle that contains a Composition corresponding to the CCD
+	 *         document and all other resources that are referenced within the
+	 *         Composition.
+	 */
+	public Bundle transformDocument(ContinuityOfCareDocument cda, Provenance provenance) {
```

Comment:
In an effort to keep the transformation contained in the transformation function, I'd think we would want the provenance created inside of the CCDTransformer instead of passed to it. Could we potentially have a generateProvenance function that creates a provenance resource given the CDA document provided to the transformDocument function? The organization information/other required info could be provided to the CCD transformer's constructor. @au2 What do you think?

Reply:
Ad I mentioned in the other PR, you could alternatively provide the provenance information to the CCDTransformer using a function instead of a constructor.

Reply:
@5t33 so question on that. As I understood what @mmccall said, we're to generate the `Provenance` object in Higgs, have it include only our agent information, then pass it to CDA2FHIR. At that point, the references in `Bundle` would be catalogued and added to `Provenance` before that would be stored in the `Bundle`.